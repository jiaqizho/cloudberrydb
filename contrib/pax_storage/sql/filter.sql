set pax_enable_filter = on;
create table pax_test.null_test_t(a int, b int, c text) using pax;

insert into pax_test.null_test_t(a) select null from generate_series(1,2)i;
insert into pax_test.null_test_t select 1, i, 'cc_' || i from generate_series(1,2)i;

insert into pax_test.null_test_t select 1, i, null from generate_series(1,2)i;

select * from pax_test.null_test_t t where t.* is null;
select * from pax_test.null_test_t t where t.* is not null;

drop table pax_test.null_test_t;
reset pax_enable_filter;
