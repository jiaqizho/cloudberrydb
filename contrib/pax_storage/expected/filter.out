set pax_enable_filter = on;
create table pax_test.null_test_t(a int, b int, c text) using pax;
insert into pax_test.null_test_t(a) select null from generate_series(1,2)i;
insert into pax_test.null_test_t select 1, i, 'cc_' || i from generate_series(1,2)i;
insert into pax_test.null_test_t select 1, i, null from generate_series(1,2)i;
select * from pax_test.null_test_t t where t.* is null;
 a | b | c 
---+---+---
   |   | 
   |   | 
(2 rows)

select * from pax_test.null_test_t t where t.* is not null;
 a | b |  c   
---+---+------
 1 | 1 | cc_1
 1 | 2 | cc_2
(2 rows)

drop table pax_test.null_test_t;
reset pax_enable_filter;
